web/src/refresh-components/Modal.tsx

```diff
@@ -0,0 +1,539 @@
+    [hasModifiedInputs, hasAttemptedClose]
+  );
+
+  return (
+    <ModalContext.Provider
+      value={{ closeButtonRef, hasAttemptedClose, setHasAttemptedClose }}
+    >
+      <ModalPortal>
+        <ModalOverlay />
+        <DialogPrimitive.Content
+          ref={(node) => {
+            // Handle forwarded ref
+            if (typeof ref === "function") {
+              ref(node);
+            } else if (ref) {
+              ref.current = node;
+            }
+            // Handle content ref with event listener
+            contentRef(node);
+          }}
+          className={cn(
+            "fixed left-[50%] top-[50%] z-[2001] translate-x-[-50%] translate-y-[-50%]",
+            "bg-background-tint-00 border rounded-16 shadow-2xl",
+            "flex flex-col overflow-hidden",
+            "data-[state=open]:animate-in data-[state=closed]:animate-out",
+            "data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0",
+            "data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95",
+            "data-[state=closed]:slide-out-to-left-1/2 data-[state=closed]:slide-out-to-top-[48%]",
+            "data-[state=open]:slide-in-from-left-1/2 data-[state=open]:slide-in-from-top-[48%]",
+            "duration-200",
+            // Size variants
+            size && sizeClassNames[size],
+            className
+          )}
+          onOpenAutoFocus={(e) => {
+            // Reset typing detection when modal opens
+            resetState();
+            props.onOpenAutoFocus?.(e);
+          }}
+          onCloseAutoFocus={(e) => {
+            // Reset typing detection when modal closes
+            resetState();
+            props.onCloseAutoFocus?.(e);
+          }}
+          onEscapeKeyDown={handleInteractOutside}
+          onPointerDownOutside={handleInteractOutside}
+          {...props}
+        >
+          {children}
+        </DialogPrimitive.Content>
+      </ModalPortal>
+    </ModalContext.Provider>
+  );
+});
+ModalContent.displayName = DialogPrimitive.Content.displayName;
+
+/**
+ * Modal Header Component
+ *
+ * Container for header content with optional bottom shadow.
+ * Use with Modal.Icon, Modal.Title, Modal.Description, and custom children.
+ *
+ * @example
+ * ```tsx
+ * <Modal.Header className="p-4" withBottomShadow>
+ *   <Modal.Icon icon={SvgWarning} />
+ *   <Modal.Title>Confirm Action</Modal.Title>
+ *   <Modal.Description>Are you sure?</Modal.Description>
+ * </Modal.Header>
+ *
+ * // With custom content
+ * <Modal.Header className="bg-background-tint-01 p-6" withBottomShadow>
+ *   <Modal.Icon icon={SvgFile} />
+ *   <Modal.Title>Select Files</Modal.Title>
+ *   <InputTypeIn placeholder="Search..." />
+ * </Modal.Header>
+ * ```
+ */
+interface ModalHeaderProps extends React.HTMLAttributes<HTMLDivElement> {
+  withBottomShadow?: boolean;
+}
+
+const ModalHeader = React.forwardRef<HTMLDivElement, ModalHeaderProps>(
+  ({ withBottomShadow = false, className, children, ...props }, ref) => {
```

Comment:
What's the purpose of "withBottomShadow"? Seems like that is something that should be controlled by scrollable divs, not the modal container itself, no?